src/internal/operators/buffer.ts

```diff
@@ -43,19 +44,38 @@ import { OperatorSubscriber } from './OperatorSubscriber';
  */
 export function buffer<T>(closingNotifier: Observable<any>): OperatorFunction<T, T[]> {
   return operate((source, subscriber) => {
+    // The current buffered values. If this is null, it's because the
+    // closingNotifier has completed before the source.
```

Comment:
nitpick: I'm not sure about this comment. AFAICT, it's nulled when the `buffer` operator's subscriber is torn down - not the subscriber to the notiffier.

Reply:
Ugh... comment cruft. Thanks for catching that.